src/utils/fhirTypes.ts

```diff
@@ -0,0 +1,235 @@
+/**
```

Comment:
Unfortunately yes, they are. The types the fhirclient library provides are accurate for the FHIR it produces. Unfortunately, this means that it indicates that some fields are always present that in truth are not required in the FHIR standard, and these fields being marked required breaks some of the code involved in generating the bundle. (Also I think it turns out that some of its FHIR types are in fact wrong or at the very least incomplete. Of course, the types in this file are also intentionally incomplete, but they're accurate for the parts of the standard we're using. And no, Partial<T> isn't a great solution, because the wrong types are child types of other types, so it ends up becoming a lot of `as unknown as Type` to jam them into the parent objects if you go that way.)